optuna_dashboard/_preference_setting.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from optuna.storages import BaseStorage
+
+from .preferential._study import PreferentialStudy
+
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+    OUTPUT_COMPONENT_TYPE = Literal["Note", "Artifact"]
+
+_SYSTEM_ATTR_FEEDBACK_COMPONENT = "preference:component"
+
+
+def _register_preference_feedback_component_type(
+    study_id: int,
+    storage: BaseStorage,
+    component_type: OUTPUT_COMPONENT_TYPE,
+    artifact_key: str = "",
+) -> None:
+    storage.set_study_system_attr(
+        study_id=study_id,
+        key=_SYSTEM_ATTR_FEEDBACK_COMPONENT,
+        value={
+            "type": component_type,
+            "artifact_key": artifact_key,
+        },
+    )
+
+
+def register_preference_feedback_component_type(
+    study: PreferentialStudy,
+    component_type: OUTPUT_COMPONENT_TYPE,
+    artifact_key: str = "",
+) -> None:
```

Comment:
If the user specifies `component_type="Artifact"` without setting the `artifact_key`, the current behavior is to set an empty string to `artifact_key`. As a result, the artifact will not be shown, and the user may not know what's wrong.

How about setting the default argument of `artifact_key` to `None` and perform a check here?